exchanges/kraken/kraken_test.go

```diff
@@ -205,530 +166,383 @@ func TestFuturesBatchOrder(t *testing.T) {
+	assert.NoError(t, err)
 }
 
 // TestGetTradesHistory API endpoint test
 func TestGetTradesHistory(t *testing.T) {
 	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, k)
 	args := GetTradesHistoryOptions{Trades: true, Start: "TMZEDR-VBJN2-NGY6DX", End: "TVRXG2-R62VE-RWP3UW"}
 	_, err := k.GetTradesHistory(context.Background(), args)
-	if err == nil {
-		t.Error("GetTradesHistory() Expected error")
-	}
+	assert.NoError(t, err)
 }
 
 // TestQueryTrades API endpoint test
 func TestQueryTrades(t *testing.T) {
 	t.Parallel()
-	_, err := k.QueryTrades(context.Background(),
-		true, "TMZEDR-VBJN2-NGY6DX", "TFLWIB-KTT7L-4TWR3L", "TDVRAH-2H6OS-SLSXRX")
-	if err == nil {
-		t.Error("QueryTrades() Expected error")
-	}
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, k)
+	_, err := k.QueryTrades(context.Background(), true, "TMZEDR-VBJN2-NGY6DX", "TFLWIB-KTT7L-4TWR3L", "TDVRAH-2H6OS-SLSXRX")
+	assert.NoError(t, err)
 }
 
 // TestOpenPositions API endpoint test
 func TestOpenPositions(t *testing.T) {
 	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, k)
 	_, err := k.OpenPositions(context.Background(), false)
-	if err == nil {
-		t.Error("OpenPositions() Expected error")
-	}
+	assert.NoError(t, err)
 }
 
 // TestGetLedgers API endpoint test
+// TODO: Needs a positive test
 func TestGetLedgers(t *testing.T) {
 	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, k)
+
 	args := GetLedgersOptions{Start: "LRUHXI-IWECY-K4JYGO", End: "L5NIY7-JZQJD-3J4M2V", Ofs: 15}
 	_, err := k.GetLedgers(context.Background(), args)
-	if err == nil {
-		t.Error("GetLedgers() Expected error")
-	}
+	assert.ErrorContains(t, err, "EQuery:Unknown asset pair", "GetLedger should error on imaginary ledgers")
 }
 
 // TestQueryLedgers API endpoint test
 func TestQueryLedgers(t *testing.T) {
 	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, k)
 	_, err := k.QueryLedgers(context.Background(), "LVTSFS-NHZVM-EXNZ5M")
-	if err == nil {
-		t.Error("QueryLedgers() Expected error")
-	}
+	assert.NoError(t, err)
 }
 
 // TestGetTradeVolume API endpoint test
 func TestGetTradeVolume(t *testing.T) {
 	t.Parallel()
-	cp, err := currency.NewPairFromString("OAVY7T-MV5VK-KHDF5X")
-	if err != nil {
-		t.Error(err)
-	}
-	_, err = k.GetTradeVolume(context.Background(), true, cp)
-	if err == nil {
-		t.Error("GetTradeVolume() Expected error")
-	}
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, k)
+	_, err := k.GetTradeVolume(context.Background(), true, bcheurPair)
+	assert.NoError(t, err, "GetTradeVolume should not error")
 }
 
-// TestAddOrder API endpoint test
-func TestAddOrder(t *testing.T) {
+// TestOrders Tests AddOrder and CancelExistingOrder
+func TestOrders(t *testing.T) {
 	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, k, canManipulateRealOrders)
+
+	// EGeneral:Invalid arguments:volume
+	t.Skip("AddOrder test is a known failure due to volume parameter")
```

Comment:
Is this due to min order amounts?

Reply:
Yes, good point. I was trying not to get distracted.
But if not that, then it'd be the price, unless we switch to buy.
I've switched to a 9k buy and that's viable.

Fixed fb030b26